src/clr-angular/forms/datepicker/date-input.ts

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {
+    AfterViewInit,
+    Directive,
+    ElementRef,
+    EventEmitter,
+    HostBinding,
+    HostListener,
+    Input,
+    OnDestroy,
+    OnInit,
+    Optional,
+    Output,
+    Renderer2,
+    Self,
+    ViewContainerRef
+} from "@angular/core";
+import {NgControl} from "@angular/forms";
+import {Subscription} from "rxjs/Subscription";
+
+import {WrappedFormControl} from "../common/wrapped-form-control";
+
+import {ClrDateContainer} from "./date-container";
+import {DateIOService} from "./providers/date-io.service";
+import {DateNavigationService} from "./providers/date-navigation.service";
+import {DatepickerEnabledService} from "./providers/datepicker-enabled.service";
+
+@Directive({selector: "[clrDate]", host: {"[class.date-input]": "true"}})
+export class ClrDateInput extends WrappedFormControl<ClrDateContainer> implements OnInit, AfterViewInit, OnDestroy {
+    /**
+     * Subscriptions to all the services and queries changes
+     */
+    private _subscriptions: Subscription[] = [];
+
+    constructor(@Optional() private container: ClrDateContainer, vcr: ViewContainerRef, private elRef: ElementRef,
+                private renderer: Renderer2, @Self() @Optional() private _ngControl: NgControl,
+                @Optional() private _dateIOService: DateIOService,
+                @Optional() private _dateNavigationService: DateNavigationService,
+                @Optional() private _datepickerEnabledService: DatepickerEnabledService) {
+        super(ClrDateContainer, vcr);
+    }
+
+    /**
+     * 1. Populate services if the date container is not present.
+     * 2. Initialize Subscriptions.
+     * 3. Process User Input.
+     */
+    ngOnInit() {
+        super.ngOnInit();
+        if (!this.container) {
+            this.populateContainerServices();
+        }
+        this.initializeSubscriptions();
+        this.processDate(this.dateValueOnInitialLoad);
```

Comment:
Can remove initial load and date value by setting a string on the input.

Reply:
Actual what we forgot over here when we discussed it yesterday was that the setter is called before `ngOnInit`, so I still have to store the initial value passed by the user. So I could repeat the writing to the input field logic but I have to store the initial value passed by the user until the services load. For now, I am just going to keep what we have as it is and not make any changes here.

cc: @youdz tagging you so that you read this comment.